x/concentrated-liquidity/fees_test.go

```diff
@@ -1015,3 +1020,152 @@ func (s *KeeperTestSuite) TestPreparePositionAccumulator() {
+	}
+
+	// Setup narrow range position across three of four accounts
+	for i := 0; i < 3; i++ {
+		s.SetupDefaultPositionAcc(clPool.GetId(), s.TestAccs[i])
+	}
+
+	// Setup consecutive range position (in relation to narrow range position) across two of four accounts
+	for i := 0; i < 2; i++ {
+		s.SetupConsecutiveRangePositionAcc(clPool.GetId(), s.TestAccs[i])
+	}
+
+	// Setup overlapping range position (in relation to narrow range position) on one of four accounts
+	s.SetupOverlappingRangePositionAcc(clPool.GetId(), s.TestAccs[0])
+
+	// Swap multiple times USDC for ETH, therefore increasing the spot price
+	ticksActivatedAfterEachSwap, totalFeesExpected := s.swapAndTrackXTimesInARow(clPool.GetId(), DefaultCoin1, ETH, cltypes.MaxSpotPrice, 7)
+	s.CollectAndAssertFees(s.Ctx, clPool.GetId(), totalFeesExpected, [][]sdk.Int{ticksActivatedAfterEachSwap}, onlyUSDC)
+
+	// Swap multiple times ETH for USDC, therefore decreasing the spot price
+	ticksActivatedAfterEachSwap, totalFeesExpected = s.swapAndTrackXTimesInARow(clPool.GetId(), DefaultCoin0, USDC, cltypes.MinSpotPrice, 7)
+	s.CollectAndAssertFees(s.Ctx, clPool.GetId(), totalFeesExpected, [][]sdk.Int{ticksActivatedAfterEachSwap}, onlyETH)
+
+	// Do the same swaps as before, however this time we collect fees after both swap directions are complete.
+	ticksActivatedAfterEachSwapUp, totalFeesExpectedUp := s.swapAndTrackXTimesInARow(clPool.GetId(), DefaultCoin1, ETH, cltypes.MaxSpotPrice, 7)
+	ticksActivatedAfterEachSwapDown, totalFeesExpectedDown := s.swapAndTrackXTimesInARow(clPool.GetId(), DefaultCoin0, USDC, cltypes.MinSpotPrice, 7)
+	totalFeesExpected = totalFeesExpectedUp.Add(totalFeesExpectedDown...)
+
+	// We expect all positions to have both denoms in their fee accumulators except USDC for the overlapping range position since
+	// it was not activated during the USDC -> ETH swap direction but was activated during the ETH -> USDC swap direction.
+	ticksActivatedAfterEachSwapTest := [][]sdk.Int{ticksActivatedAfterEachSwapUp, ticksActivatedAfterEachSwapDown}
+	denomsExpected := [][]string{{USDC, ETH}, {USDC, ETH}, {USDC, ETH}, {"", ETH}}
+
+	s.CollectAndAssertFees(s.Ctx, clPool.GetId(), totalFeesExpected, ticksActivatedAfterEachSwapTest, denomsExpected)
+}
+
+// CollectAndAssertFees collects fees from a given pool for all positions and verifies that the total fees collected match the expected total fees.
+// The method also checks that if the ticks that were active during the swap lie within the range of a position, then the position's fee accumulators
+// are not empty. The total fees collected are compared to the expected total fees within an additive tolerance defined by an error tolerance struct.
+func (s *KeeperTestSuite) CollectAndAssertFees(ctx sdk.Context, poolId uint64, totalFees sdk.Coins, activeTicks [][]sdk.Int, expectedFeeDenoms [][]string) {
+	var totalFeesCollected sdk.Coins
+	// Claim full range position fees across all four accounts
+	for i := 0; i < 4; i++ {
```

Comment:
Let's make these parameters that we configure at the top of the test, allowing all helpers take these parameters as inputs. I think this will be especially helpful if we start adding more functional tests. For example, LP + Swap + Collect Fees and various other story-driven flows

Reply:
https://github.com/osmosis-labs/osmosis/pull/4282/commits/9cd236692a27225f65cd8f66adbdf5a96e014fa4